modules/core/src/parallel.cpp

```diff
@@ -789,12 +838,37 @@ int cv::getNumberOfCPUs(void)
     GetSystemInfo( &sysinfo );
 #endif
 
-    return (int)sysinfo.dwNumberOfProcessors;
-#elif defined __ANDROID__
-    static int ncpus = getNumberOfCPUsImpl();
+    return concurentThreadsSupported == 0 ? (unsigned)sysinfo.dwNumberOfProcessors : (concurentThreadsSupported < (unsigned)sysinfo.dwNumberOfProcessors ? concurentThreadsSupported : (unsigned)sysinfo.dwNumberOfProcessors);
+#elif defined __linux__ || defined __GLIBC__ || defined __HAIKU__ || defined __EMSCRIPTEN__ || defined __ANDROID__
+    #if defined __ANDROID__
+        static unsigned ncpus_impl = (unsigned)getNumberOfCPUsImpl("/sys/devices/system/cpu/possible");
+    #else /* Linux */
+        unsigned ncpus_impl_cpuset = (unsigned)getNumberOfCPUsImpl("/sys/fs/cgroup/cpuset/cpuset.cpus");
+        unsigned ncpus_impl_cfs = getNumberOfCPUsCFS();
+        static unsigned ncpus_impl;
+        if(ncpus_impl_cpuset == 0 && ncpus_impl_cfs == 0)
+            ncpus_impl = 1;
+        else if(!ncpus_impl_cpuset != !ncpus_impl_cfs)          /* One of them is zero */
+            ncpus_impl = ncpus_impl_cpuset | ncpus_impl_cfs;    /* Take non zero value */
+        else                                                    /* Both non zero, so take min*/
+            ncpus_impl = ncpus_impl_cpuset < ncpus_impl_cfs ? ncpus_impl_cpuset : ncpus_impl_cfs;
+
+        if(concurentThreadsSupported == 0) { /* somehow hardware_concurrency failed or c++ version < 11 */
+            cpu_set_t cpu_set;
+            sched_getaffinity(0, sizeof(cpu_set), &cpu_set);
```

Comment:
Looks like this API is only available on Linux and only in case when `_GNU_SOURCE` is enabled: https://linux.die.net/man/2/sched_getaffinity

Function call should be checked for errors.

On my platform `std::thread::hardware_concurrency` returns 8, thus affinity check is effectively disabled.

Reply:
I seem to have missed this, I'll add it